### PR TITLE
Disable rx checksumming by default (bnc#910298)

### PR DESCRIPTION
**Pull request by dirkmueller**

### CODE REVIEW

**Comment by tboerger** [comment]
Do we want to merge that before next milestone?

File: chef/cookbooks/network/templates/default/suse-cfg.erb [hunk 1]
```diff
@@ -20,8 +20,8 @@ BOOTPROTO=none
 <% else -%>
 BOOTPROTO=static
   <% unless @nic.kind_of?(Nic::Vlan) or @nic.kind_of?(Nic::Bond) -%>
-    <% unless @enable_tx_offloading -%>
-ETHTOOL_OPTIONS="-K iface tx off"
+    <% if @ethtool_options -%>
```
Comment (vuntz) [inline chef/cookbooks/network/templates/default/suse-cfg.erb]: You can't use a string as a boolean the way you expect it:

```
puts "ok" if ""
(irb):8: warning: string literal in condition
ok
```

You need to check if the string is empty or not.

Comment (tboerger) [inline chef/cookbooks/network/templates/default/suse-cfg.erb]: ```unless @ethtool_options.emtpy?``` should do it.

File: chef/cookbooks/network/templates/default/suse-cfg.erb [hunk 2]
```diff
@@ -20,8 +20,8 @@ BOOTPROTO=none
 <% else -%>
 BOOTPROTO=static
   <% unless @nic.kind_of?(Nic::Vlan) or @nic.kind_of?(Nic::Bond) -%>
-    <% unless @enable_tx_offloading -%>
-ETHTOOL_OPTIONS="-K iface tx off"
+    <% unless @ethtool_options.empty? -%>
+ETHTOOL_OPTIONS="-K iface <%= @ethtool_options -%>"
```
Comment (rhafer) [inline chef/cookbooks/network/templates/default/suse-cfg.erb]: I am not quite sure, but doesn't this need to be ```@ethtool_options.join(" ")```?

**Comment by rhafer** [comment]
Another small nitpick: You forgot to address @aspiers' comment about the full bugzilla url.

File: chef/cookbooks/network/templates/default/suse-cfg.erb (hunk 2 above)
Comment (aspiers) [inline chef/cookbooks/network/templates/default/suse-cfg.erb]: No, that's done already by line 414:

    ethtool_options = ethtool_options.join(" ")

**Comment by aspiers** [comment]
Thank you :)

File: chef/cookbooks/network/templates/default/suse-cfg.erb (hunk 2 above)
Comment (rhafer) [inline chef/cookbooks/network/templates/default/suse-cfg.erb]: Argh. Sorry, overlooked that somehow :/

**Comment by dirkmueller** [comment]
@rhafer, @aspiers I don't think the full bugzilla url is really needed, but if it makes you happy, here it is ;)


I think its more important to have the bnc# tagging in the commit summary so that it automatically ends up in the .changes file

**Comment by aspiers** [comment]
Both are important - `bsc#` tag for the `.changes` file as you say, and the URL so that a) it's clear for outsiders who have no idea what `bsc#` or `bnc#` mean, and b) so that we can easily open the bug from git history with a single click rather than having to type anything.

This was [discussed and agreed as best practice in previous team workshops](https://etherpad.nue.suse.com/p/cloud-pull-requests).

File: chef/data_bags/crowbar/migrate/network/010_add_rx_offloading.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+def downgrade ta, td, a, d
+  unless ta.has_key? "enable_tx_offloading"
+    a.delete "enable_tx_offloading"
+  end
+  unless ta.has_key? "enable_rx_offloading"
+    a.delete "enable_rx_offloading"
+  end
+  return a, d
+end
+
```
Comment (vuntz) [inline chef/data_bags/crowbar/migrate/network/010_add_rx_offloading.rb]: No need for the last empty line (or two lines between the two methods).

But this is nit-picky, so +1 and feel free to merge without fixing this :-)

**Pull request merged**